assets/js/admin/course-theme-sidebar.js

```diff
@@ -3,49 +3,147 @@
  */
 import { PluginDocumentSettingPanel } from '@wordpress/edit-post';
 import { __ } from '@wordpress/i18n';
-import { SelectControl } from '@wordpress/components';
+import { ToggleControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import useCourseMeta from '../../react-hooks/use-course-meta';
+import { name as courseOutlineBlockName } from '../../blocks/course-outline/outline-block/block.json';
+import { name as courseModuleBlockName } from '../../blocks/course-outline/module-block/block.json';
+import { name as courseLessonBlockName } from '../../blocks/course-outline/lesson-block/block.json';
 
 const SENSEI_THEME = 'sensei-theme';
 const WORDPRESS_THEME = 'wordpress-theme';
+const SENSEI_PREVIEW_QUERY = 'sensei_theme_preview';
+
+const canPreview = ( block ) =>
+	block.name === courseLessonBlockName &&
+	block.attributes.id &&
+	! block.attributes.draft;
 
 /**
  * Course Theme Sidebar component.
  */
 const CourseThemeSidebar = () => {
 	const [ theme, setTheme ] = useCourseMeta( '_course_theme' );
 
+	const globalLearningModeEnabled =
+		window.senseiSettings.sensei_learning_mode_all || false;
+	const currentPost = useSelect( ( select ) =>
+		select( 'core/editor' ).getCurrentPost()
+	);
+
+	const firstLesson = useSelect( ( select ) => {
+		const { getBlocks, getBlockAttributes } = select( 'core/block-editor' );
+		const blocks = getBlocks();
+		const courseOutline = blocks.find(
```

Comment:
I think if we have the outline as a child of another block, like the Conditional Content block, for example, it wouldn't work, right? Maybe `getFirstBlockByName` from `assets/blocks/course-outline/data.js` can help you with that?

Reply:
Nice catch! Updated. 2cb0011a6840bd30a5b76473c71a031b97933a92
Thanks for the `getFirstBlockByName` method!